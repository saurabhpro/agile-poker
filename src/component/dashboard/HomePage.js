import React from 'react';

import SignIn from './signin/SignIn';

export default function HomePage() {
  return (
    <div>
      <SignIn />
    </div>
  );
}
