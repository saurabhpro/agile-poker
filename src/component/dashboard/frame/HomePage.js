import React from 'react';

import { Row } from 'react-bootstrap';

import CardPanel from './CardPanel';
import RightSidePanel from './RightSidePanel';
import { getCurrentlyLoggedInUserDetails } from '../../utils/firebaseDb';

const HomePage = ({ userName }) => {
  const [currentUser, setCurrentUser] = React.useState();

  React.useEffect(() => {
    const getCurrentUser = () => {
      getCurrentlyLoggedInUserDetails(userName).then((usr) =>
        setCurrentUser(usr),
      );
    };

    if (!currentUser) {
      getCurrentUser();
    }
  }, [currentUser, userName]);

  return (
    <div
      className="container-md content-justify-center-sm"
      style={{ flexGrow: 3 }}
    >
      <Row>
        <div className="col-sm-9">
          <CardPanel currentUser={currentUser}/>
        </div>
        <div className="col-md">
          <RightSidePanel currentUser={currentUser} />
        </div>
      </Row>
    </div>
  );
};

export default HomePage;
