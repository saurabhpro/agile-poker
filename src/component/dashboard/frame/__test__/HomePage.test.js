import React from 'react';
import { shallow } from 'enzyme';
import HomePage from '../HomePage';

let component;

describe('main center page after successful login', () => {
  beforeAll(() => {
    component = shallow(<HomePage userName={'Saurabh'} />);
    console.debug(component.debug());
  });

  it('matches the rendered snapshot', ()
   => {
    expect(component).toMatchSnapshot();
  });

  it('should render one <CardPanel>', () => {
    expect(component.find('CardPanel')).toHaveLength(0);
  });

  it('should render one <RightSidePanel>', () => {
    expect(component.find('RightSidePanel')).toHaveLength(0);
  });
});
