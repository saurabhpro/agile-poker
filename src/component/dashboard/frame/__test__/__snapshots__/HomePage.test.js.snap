// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main center page after successful login matches the rendered snapshot 1`] = `ShallowWrapper {}`;
